libs/entitlement/src/main/java/org/elasticsearch/entitlement/runtime/policy/PolicyManager.java

```diff
@@ -843,6 +803,10 @@ private static boolean isTriviallyAllowed(Class<?> requestingClass) {
             generalLogger.debug("Entitlement trivially allowed from system module [{}]", requestingClass.getModule().getName());
             return true;
         }
+        if (moduleEntitlementsCache.isAlwaysAllowed(requestingClass)) {
```

Comment:
My thinking was to _move_ the logic in isTriviallyAllowed, so that the test infra can add "is this a test class". Otherwise this additional condition is only ever exercised in tests.

Reply:
Good point. A branch that's dead in production feels wrong.

Reply:
It's a little awkward because `isTriviallyAllowed` checks `SYSTEM_LAYER_MODULES`. `PolicyManager` seems like a good home for logic that's the same in production and unit tests.

Let me try ending it with an unconditional call to `moduleEntitlementsCache` and see if you like how that looks.

Reply:
Why can't SYSTEM_LAYER_MODULES be defined inside this lookup?

Reply:
It could I guess. That seemed like part of `PolicyManager`'s responsibility, along with similar things like `DEFAULT_FILESYSTEM_CLASS` and `MODULES_EXCLUDED_FROM_SYSTEM_MODULES`, but yeah, it could go elsewhere.